### PR TITLE
Fix GridGenerator::merge_triangulations() for simplex meshes

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: source/grid/grid_generator.cc [hunk 1]
```diff
@@ -6463,7 +6463,11 @@ namespace GridGenerator
 
     // reorder the cells to ensure that they satisfy the convention for
     // edge and face directions
-    GridTools::consistently_order_cells(cells);
+    if (std::find_if(cells.begin(), cells.end(), [](const auto &cell) {
+          return cell.vertices.size() !=
+                 ReferenceCells::get_hypercube<dim>().n_vertices();
+        }) == cells.end())
+      GridTools::consistently_order_cells(cells);
```
Comment (drwells) [inline source/grid/grid_generator.cc:6470]:
```suggestion
    if (std::all_of(cells.begin(), cells.end(), [](const auto &cell) {
          return cell.vertices.size() ==
                 ReferenceCells::get_hypercube<dim>().n_vertices();
        }))
      GridTools::consistently_order_cells(cells);
```

**Review by drwells — COMMENTED** [review]
I'm traveling today so I might not be around to approve this later. This is fine but it would be better if we avoided the double negative.

File: source/grid/grid_generator.cc (hunk 1 above)
Comment (masterleinad) [reply source/grid/grid_generator.cc:6470]: I think the condition really rather is that there is something to be done if any of the cells reference is a  (generalized) hex, isn't it? Shouldn't it be
```suggestion
    if (std::any_of(cells.begin(), cells.end(), [](const auto &cell) {
          return cell.vertices.size() ==
                 ReferenceCells::get_hypercube<dim>().n_vertices();
        }) == cells.end())
      GridTools::consistently_order_cells(cells);
```
then?

Comment (peterrum) [reply source/grid/grid_generator.cc]: The right logic is `all_of`. `GridTools::consistently_order_cells` only works if all cells are hypercubes. In all other cases, we store the orientation of faces/edges anyway. I have made the modification. It ready from my side!

**Review by masterleinad — APPROVED** [review]

**Review by drwells — APPROVED** [review]

**Pull request merged**